src-ts/proposalCreator.ts

```diff
@@ -210,4 +211,41 @@ export class RoundTripProposalCreator {
       proposalDescription
     );
   }
+
+  public async createTimelockScheduleArgs(
+    l2GovConfig: L2GovConfig,
+    upgradeAddr: string,
+    description: string,
+    options?: {
+      _upgradeValue?: BigNumber,
+      _upgradeArgs?: string,
+      _delay?: BigNumber,
+      _predecessor?: string,
+    }
+  
+  )  {
+    const upgradeValue = options?._upgradeValue || BigNumber.from(0) // default to 0 value
+    const predecessor = options?._predecessor || "0x"; // default to no predecessor
+    
+    const l2Gov = await L2ArbitrumGovernor__factory.connect(  l2GovConfig.governorAddr, l2GovConfig.provider)
+    const l2TimelockAddress = await l2Gov.timelock()
+    const l2Timelock = await ArbitrumTimelock__factory.connect(l2TimelockAddress, l2GovConfig.provider)
+    const delay = options?._delay? options?._delay : await l2Timelock.getMinDelay(); // default to min delay
+
+    let ABI = [ "function perform() external" ]; // TODO: allow for optional args
```

Comment:
Remaining TODO in code  - should remove and create issue if we want to keep this